python/pyarrow/types.pxi

```diff
@@ -836,6 +836,26 @@ cdef class BaseExtensionType(DataType):
         DataType.init(self, type)
         self.ext_type = <const CExtensionType*> type.get()
 
+    def __arrow_ext_class__(self):
+        """Return an extension array class to be used for building or
+        deserializing arrays with this extension type.
+
+        This method should return a subclass of the ExtensionArray class. By
+        default, if not specialized in the extension implementation, an
+        extension type array will be a built-in ExtensionArray instance.
+        """
+        return ExtensionArray
```

Comment:
It occurred to me that perhaps we should be generating dummy ExtensionArray classes, perhaps from the `extension_name`? However, this does seem to just work.

Reply:
> However, this does seem to just work.

Yes, this works as long as you are fine with the base implementation of that class (eg it's implementation of to_numpy, etc). 
Once you want to customize that, you'll need a custom class, but I am not sure if that's straightforward to do at the moment ("register" such a custom class for extension type that is actually implemented in C++). In any case, for this PR using ExtensionArray is fine (since a dummy generated class would not give anything of additional functionality, just a different name of the class)